src/components/Calculator.vue

```diff
@@ -1,168 +1,316 @@
-      padding: 0 $environment-block-padding-small-screen;
-    }
+        methods: {
+            onClearScreen() {
+                this.calc_screen = '';
+                this.first_argument = '';
+                this.second_argument = '';
+                this.lastSecond_argument = '';
+                this.operation = '';
+                this.lastOperation = '';
+                this.error = '';
+            },
 
-    .change-calc-view {
-      width: 120px;
-      height: 25px;
-      border: none;
-      border-radius: 3px;
-      color: #fbfbfb;
-      background-color: #ff7c1f;
-    }
-  }
+            onDigitClick(digit) {
 
-  .calculator {
-    max-width: 500px;
-    margin: 0 auto;
-    padding: 20px;
+                if (this.error) {
+                    this.onClearScreen();
+                }
 
-    @media screen and (max-height: 400px) {
-      overflow-y: scroll;
-    }
+                const number = this.operation ? this.second_argument : this.first_argument;
 
-    .calc-screen {
-      padding-right: 10px;
-      border: solid 2px #4d90fd;
-      border-radius: 4px;
-      text-align: right;
+                if (number === '0') {
+                    this.operation ? this.second_argument = '' : this.first_argument = '';
+                }
 
-      strong {
-        font-size: 2em;
-        letter-spacing: 2px;
-      }
-    }
+                if (this.operation) {
+                    this.second_argument += digit;
+                } else {
+                    this.first_argument += digit;
+                }
+                this.result();
+            },
 
-    .two-horizontal-cells {
-      grid-area: large_button;
-    }
+            onDotClick() {
+                let regExpForDot = /\./;
+
+                if (!this.operation) {
+                    this.first_argument += !this.first_argument ? '0' : '';
+
+                    this.first_argument += this.isValueExist(this.first_argument, regExpForDot) != -1 ? '' : '.';
+                } else {
+                    this.second_argument += !this.second_argument ? '0' : '';
+                    this.second_argument += this.isValueExist(this.second_argument, regExpForDot) != -1 ? '' : '.';
+                }
+
+                this.result();
+            },
+
+            onOperationClick(operation) {
+
+                if (!this.first_argument) {
+                    this.first_argument = '0';
+                }
+
+                this.first_argument = this.deleteDotIfNoDigitAfter(this.first_argument)
+
+                if (this.second_argument) {
+                    this.second_argument = this.deleteDotIfNoDigitAfter(this.second_argument)
+                    this.onGetResult();
+                }
+
+                this.operation = operation;
+                this.result();
+            },
+
+            deleteDotIfNoDigitAfter(argument) {
+
+                if (this.isValueExist(argument, /\./) == (argument.length - 1)) {
```

Comment:
argument.indexOf('`') is easier to read then `this.isValueExist(argument, /\./)`

Also use '===' instead of '=='.

Reply:
ok, deleted reg. exp.